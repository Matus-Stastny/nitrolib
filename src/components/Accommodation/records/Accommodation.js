// @flow strict

import type { RoomType } from "../../../records/Room";

export type AccommodationType = {|
  hotelName: string,
  rating: number,
  location: {|
    latitude: number,
    longitude: number,
  |},
  address: string,
  photoUrl: string,
  rooms: Array<RoomType>,
  mapboxToken: string,
  zoom?: number,
  isMMB: boolean,
|};
