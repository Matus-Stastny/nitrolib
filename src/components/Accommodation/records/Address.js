// @flow strict

export type AddressType = ?{|
  +fullAddress: ?string,
|};
