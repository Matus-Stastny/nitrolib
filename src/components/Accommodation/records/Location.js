// @flow strict

export type LocationType = {|
  longitude: number,
  latitude: number,
  address: string,
  hotelName: string,
  mapboxToken: string,
  desktopWidth?: number,
  zoom?: number,
|};
