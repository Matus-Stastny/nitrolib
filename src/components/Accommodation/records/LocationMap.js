// @flow strict

export type LocationType = {|
  center: {|
    latitude: number,
    longitude: number,
  |},
  label: string,
  zoom: number,
  desktopWidth: number,
  mapboxToken: string,
|};
