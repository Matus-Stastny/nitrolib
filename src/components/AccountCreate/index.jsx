// @flow strict
import * as React from "react";
import styled from "styled-components";
import OrbitText from "@kiwicom/orbit-components/lib/Text";
import Heading from "@kiwicom/orbit-components/lib/Heading";
import InputField from "@kiwicom/orbit-components/lib/InputField";
import Button from "@kiwicom/orbit-components/lib/Button";
import Illustration from "@kiwicom/orbit-components/lib/Illustration";

import { Consumer } from "../../services/intl/context";
import Text from "../Text";

const SpacingXSmall = styled.div`
  margin-bottom: 8px;
`;

const SpacingMedium = styled.div`
  margin-bottom: 16px;
`;

type Props = {|
  email: string,
  password: string,
  passwordConfirm: string,
  onEmailChange: (ev: SyntheticEvent<HTMLInputElement>) => void,
  onPasswordChange: (ev: SyntheticEvent<HTMLInputElement>) => void,
  onPasswordConfirmChange: (ev: SyntheticEvent<HTMLInputElement>) => void,
  onContinue: (ev: SyntheticEvent<HTMLButtonElement>) => void,
|};

const AccountCreate = ({
  email,
  onEmailChange,
  onContinue,
  password,
  passwordConfirm,
  onPasswordChange,
  onPasswordConfirmChange,
}: Props) => (
  <Consumer>
    {intl => (
      <React.Fragment>
        <Illustration name="EnjoyApp" size="small" />
        <SpacingXSmall>
          <Heading element="h2">
            <Text t={__("account.create_account")} />
          </Heading>
        </SpacingXSmall>
        <SpacingXSmall>
          <Text t={__("account.create_account_description")} />
        </SpacingXSmall>
        <SpacingMedium>
          <InputField
            label={intl.translate(__("account.email"))}
            placeholder={intl.translate(__("account.email_placeholder"))}
            type="email"
            value={email}
            onChange={onEmailChange}
          />
        </SpacingMedium>
        <SpacingMedium>
          <InputField
            label={intl.translate(__("account.password"))}
            type="password"
            value={password}
            onChange={onPasswordChange}
          />
        </SpacingMedium>
        <SpacingMedium>
          <InputField
            label={intl.translate(__("account.password_confirmaiton"))}
            type="password"
            value={passwordConfirm}
            onChange={onPasswordConfirmChange}
          />
        </SpacingMedium>
        <SpacingMedium>
          <Button onClick={onContinue}>
            <Text t={__("account.create")} />
          </Button>
        </SpacingMedium>
        <OrbitText size="small">
          <Text t={__("account.terms_and_privacy_policy")} html />
        </OrbitText>
      </React.Fragment>
    )}
  </Consumer>
);

export default AccountCreate;
