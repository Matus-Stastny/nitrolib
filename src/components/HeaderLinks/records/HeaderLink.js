// @flow strict
export type HeaderLink = {|
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
|};
