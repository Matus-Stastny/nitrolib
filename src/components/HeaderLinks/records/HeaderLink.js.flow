// @flow strict
export type ActiveTab = "travel" | null;

export type HeaderLink = {|
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
|};

export type SearchForm = {|
  mode: string,
  destination: { type: string, name: string }, // TODO move proper search form into records @vacuum
  checkIn: Date,
  checkOut: Date | null,
  adults: number,
  children: number,
|};
