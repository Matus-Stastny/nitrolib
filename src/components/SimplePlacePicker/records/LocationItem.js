// @flow strict
type Params = {|
  id: string,
  name: string,
  slug: string,
  code: string,
|};

type Gps = {|
  lat: number,
  lon: number,
|};

type Tag = {|
  month_from: number,
  month_to: number,
  tag: string,
|};

type City = {|
  autonomous_territory: boolean | null,
  code: string,
  continent: Params,
  country: Params,
|};

type Region = {|
  id: string,
  name: string,
  slug: string,
|};

type AlternativePoints = {|
  id: string,
  distance: number,
  duration: number,
|};

export type LocationType = "country" | "city" | "airport" | "continent" | "special";

export type LocationItem = {|
  active: boolean,
  alternative_departure_points: Array<AlternativePoints>,
  alternative_names: Array<string>,
  autonomous_territory: boolean | null,
  bus_stations?: number,
  city?: City,
  code: string,
  currency?: string,
  organizations?: Array<string>,
  neighbours?: Array<string>,
  dst_popularity_score: number,
  global_rank_dst: number,
  icao?: string,
  hotels: number,
  continent?: Params,
  country?: Params,
  population?: number,
  id: string,
  int_id?: string,
  location: Gps,
  name: string,
  rank: number,
  stations?: number,
  region: Region,
  slug?: string,
  code?: string,
  subdivision?: Params,
  airports?: number,
  special?: Array<mixed>,
  tags: Array<Tag>,
  timezone: string,
  type: LocationType,
|};
