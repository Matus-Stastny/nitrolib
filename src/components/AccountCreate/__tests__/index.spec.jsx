// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import AccountCreate from "..";

import { intlDefault } from "../../../records/Intl";

describe("#AccountCreate", () => {
  test("render", () => {
    const wrapper = shallow(
      <AccountCreate
        email=""
        password=""
        passwordConfirm=""
        onEmailChange={() => {}}
        onPasswordChange={() => {}}
        onPasswordConfirmChange={() => {}}
        onContinue={() => {}}
      />,
    );

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });

  test("validation errors", () => {
    const wrapper = shallow(
      <AccountCreate
        email=""
        password=""
        passwordConfirm=""
        onEmailChange={() => {}}
        onPasswordChange={() => {}}
        onPasswordConfirmChange={() => {}}
        onContinue={() => {}}
        emailError="Wrong format of e-mail"
        passwordError="Password should be more complex"
        passwordConfirmError="Passwords do not match"
      />,
    );

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });
});
