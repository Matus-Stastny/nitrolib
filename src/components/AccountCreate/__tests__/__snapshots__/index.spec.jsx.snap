// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#AccountCreate render 1`] = `
<AccountCreate__Wrapper>
  <ModalHeader>
    <Illustration
      name="EnjoyApp"
      size="small"
    />
    <Heading
      element="h2"
    >
      <Text
        html={false}
        t="account.create_account"
        values={Object {}}
      />
    </Heading>
    <Text
      html={false}
      t="account.create_account_description"
      values={Object {}}
    />
  </ModalHeader>
  <ModalSection>
    <Stack
      desktop={
        Object {
          "spacing": "comfy",
        }
      }
    >
      <InputField
        label="account.email"
        onChange={[Function]}
        placeholder="account.email_placeholder"
        type="email"
        value=""
      />
      <InputField
        label="account.password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <InputField
        label="account.password_confirmaiton"
        onChange={[Function]}
        type="password"
        value=""
      />
      <Button
        onClick={[Function]}
      >
        <Text
          html={false}
          t="account.create"
          values={Object {}}
        />
      </Button>
      <Text
        size="small"
      >
        <Text
          html={true}
          t="account.terms_and_privacy_policy"
          values={Object {}}
        />
      </Text>
    </Stack>
  </ModalSection>
</AccountCreate__Wrapper>
`;
